sudo/lib/exec/monitor.rs

```diff
@@ -130,29 +148,33 @@ impl<'a> MonitorClosure<'a> {
                     // Forward signal to the command.
                     MonitorMessage::Signal(signal) => {
                         if let Some(command_pid) = self.command_pid {
-                            Self::send_signal(signal, command_pid)
+                            send_signal::<true>(signal, command_pid)
                         }
                     }
                 }
             }
         }
     }
-
-    /// Send a signal to the command
-    fn send_signal(signal: c_int, command_pid: ProcessId) {
-        // FIXME: We should call `killpg` instead of `kill`.
-        match signal {
-            SIGALRM => {
-                // Kill the command with increasing urgency. Based on `terminate_command`.
-                kill(command_pid, SIGHUP).ok();
-                kill(command_pid, SIGTERM).ok();
-                std::thread::sleep(Duration::from_secs(2));
-                kill(command_pid, SIGKILL).ok();
-            }
-            signal => {
-                // Send the signal to the command.
-                kill(command_pid, signal).ok();
-            }
+}
+/// Send a signal to the command.
+fn send_signal<const FROM_PARENT: bool>(signal: c_int, command_pid: ProcessId) {
```

Comment:
is there a reason for making `FROM_PARENT` a const generic instead of a regular argument? I personally consider turbofish syntax something that should be used sparingly and instead one should rely on type inference whenever possible.

Reply:
I like the idea of not relying in optimizations for stuff like this, on the other hand, the from_parent thing is only used under the dev feature so it might be irrelevant to use a const here